src/plugins/explore/public/application/hooks/editor_hooks/use_change_query_editor/use_change_query_editor.test.ts

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { renderHook } from '@testing-library/react-hooks';
```

Comment:
i'm worried the structure of the query panel being too coupled with explore plugin. if the goal is to make this eventually global then it should likely exist outside of the explore plugin but with this current implementation it would be difficult to extract that out in the future. which would then make that itself a larger project.

are we sure we want to say this is global accessible.

Reply:
this is (almost) globally accessible in the context of the explore app. If you look at the DOM tree both the redux and EditorContext are near the top